server/server.go

```
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+// Server : struct
+type Server struct {
+	ListenHost string
+	ListenPort string
+	Router     *mux.Router
+}
+
+func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.Router.ServeHTTP(w, r)
+}
+
+func (s *Server) myHandler(w http.ResponseWriter, r *http.Request) {
+	type response struct {
+		Host       string      `json:"host"`
+		UserAgent  string      `json:"user_agent"`
+		RequestURI string      `json:"request_uri"`
+		Headers    http.Header `json:"headers"`
+	}
+
+	newResponse := response{
+		r.Host,
+		r.UserAgent(),
+		r.RequestURI,
+		r.Header,
+	}
+	js, err := json.Marshal(newResponse)
+	if err != nil {
+		log.Fatal(err)
```

You should return an error message to the client in case of error. Don't forget about status code as well.

---

If I am not mistaken log.Fatal() will exit the program. You shouldn't stop server in case of error.